generators/javascript/math.js

```diff
@@ -147,60 +147,62 @@ Blockly.JavaScript['math_constant'] = function(block) {
 Blockly.JavaScript['math_number_property'] = function(block) {
   // Check if a number is even, odd, prime, whole, positive, or negative
   // or if it is divisible by certain number. Returns true or false.
-  const number_to_check = Blockly.JavaScript.valueToCode(block, 'NUMBER_TO_CHECK',
-      Blockly.JavaScript.ORDER_MODULUS) || '0';
-  const dropdown_property = block.getFieldValue('PROPERTY');
+  const PROPERTIES = {
+    'EVEN': [' % 2 == 0', Blockly.JavaScript.ORDER_MODULUS,
```

Comment:
Per the discussion [here](https://github.com/google/blockly/issues/5678), use `===` on this line.  The `%` operator will always result in something with a number type so using `===` is functionally equivalent to using `==` here.   As a note, the result may be `NaN`, not a number, if the left-hand side of the `%` is something that can't be coerced into a number type like `'foo'`, but `NaN` also has number type.